backend/routes/agency.js

```diff
@@ -8,38 +8,53 @@ const { Agency } = require("../models");
 const router = express.Router();
 
 /**
- * Validates the distribution start time for the given day if the day is
- * selected, using express-validator. Expects the format "HH:mm AM".
+ * Checks that the distribution start time for the given day is ISO 8601 format,
+ * if the day is selected, using express-validator.
  * @param {String} day Day of the week to check ("monday", "tuesday", etc.)
  */
 const validateDistributionStartTime = (day) =>
   body(`distributionStartTimes.${day}`)
     .trim()
     .if((value, { req }) => req.body.distributionDays[day])
-    .custom((value) => {
-      // custom time format
-      return value.match(/^(0[1-9]|1[0-2]):([0-5][0-9]) [AP]M$/);
-    });
+    .isISO8601();
+
+/**
+ * Checks that the retail rescue start time for the given day is ISO 8601
+ * format, if the day is selected, using express-validator.
+ * @param {String} day Day of the week to check ("monday", "tuesday", etc.)
+ */
+const validateRetailRescueStartTime = (day) =>
+  body(`retailRescueStartTimes.${day}`)
+    .trim()
+    .if((value, { req }) => req.body.retailRescueDays[day])
+    .isISO8601();
+
+/**
+ * Checks that the retail rescue location for the given day exists, if the day
+ * is selected, using express-validator.
+ * @param {String} day Day of the week to check ("monday", "tuesday", etc.)
+ */
+const validateRetailRescueLocation = (day) =>
+  body(`retailRescueLocations.${day}`)
+    .trim()
+    .if((value, { req }) => req.body.retailRescueDays[day])
+    .not()
+    .isEmpty();
 
 /**
  * Form Validation: validationChain is an array of expected formats for the
  * specified model fields.
  */
 const validationChain = [
   body("tableContent.agencyNumber").trim().isNumeric({ no_symbols: true }),
-  body("tableContent.name").trim().not().isEmpty(),
```

Comment:
Removed validators that were just checking for the existence of required fields, because mongoose already does that so it was unnecessary. The only validators that remain in agency.js should be those checking for specific formats or which are verifying existence only in certain conditions.